misk/src/main/kotlin/misk/config/MiskConfig.kt

```diff
@@ -34,7 +45,14 @@ object MiskConfig {
     environment: Environment,
     overrideFiles: List<File> = listOf()
   ): T {
-    val mapper = ObjectMapper(YAMLFactory()).registerModules(KotlinModule(), JavaTimeModule())
+
+    checkConfigStructure(configClass)
+
+    val simpleModule = SimpleModule()
```

Comment:
super minor language pedantry, but kotlin has a nice `apply` syntax that lets you operate on this like a builder pattern

```kotlin
val simpleModule = SimpleModule().apply { addDeserializer(Secret::class.java, SecretDeserializer()) }
```